heat/core/manipulations.py

```diff
@@ -89,6 +91,81 @@ def balance(array: DNDarray, copy=False) -> DNDarray:
 DNDarray.balance.__doc__ = balance.__doc__
 
 
+def broadcast_arrays(*arrays: DNDarray) -> List[DNDarray]:
+    """
+    Broadcasts one or more arrays against one another.
+
+    Parameters
+    ----------
+    arrays : DNDarray
+        An arbitrary number of to-be broadcasted ``DNDarray``s.
+    """
+    if len(arrays) <= 1:
+        return arrays
+
+    try:
+        arrays = sanitation.sanitize_distribution(*arrays, target=arrays[0])
+    except NotImplementedError as e:
+        raise ValueError(e)
+
+    # extract torch tensors
+    t_arrays = list(array.larray for array in arrays)
+
+    broadcasted = torch.broadcast_tensors(*t_arrays)
+
+    out = []
+    for i in range(len(broadcasted)):
+        out.append(factories.array(broadcasted[i], dtype=arrays[i].dtype, device=arrays[i].device))
```

Comment:
Maybe I got that wrong, but wasnt there another PR where we wanted to replace the usage of `factories.array` by `DNDarray` for performance reasons?

Reply:
you're right, and the split axis was missing here as well. Fixed.